ExploringOptions/Alternative_Solutions_(WASI_Wasm).md

```diff
@@ -0,0 +1,296 @@
+
+In contrast, WASI 0.2+ adoption remains limited, with only a few notable implementations:
+
+* **Wasmtime** – The primary implementation and reference runtime for the Component Model-based WASI.  
+* **Potentially WasmEdge** – some initial work towards support has begun, but full support has not yet been confirmed.  
+* **JCo** – A JavaScript toolchain designed to work with Preview 2+ WASI modules, though it is not a runtime.
+
+With WASI 1.0 expected to finalize this transition, existing WASI Preview 1-based runtimes and applications face an uncertain path forward. The lack of clear migration options risks fragmenting the ecosystem and leaving many early adopters behind.
+
+Beyond the WASI standard, there are academic and experimental implementations which have explored extending WASI Preview 1\.  Projects such as the WebAssembly Linux Interface (WALI)[^1], which explores exposing the Linux kernel APIs via WASI core wasm function imports and exports, paving the way for WASI interfaces to be implemented in a portable, layered way have also been introduced and evaluated.
+
+## Long term Toolchain Support \- Continuing the Legacy of Success
+
+A stable and long-term supported toolchain is essential for any healthy platform and WASI is no exception. Toolchains form the foundation for application portability, enabling developers to build and maintain their software.  A stable toolchain gives developers confidence that their binaries will remain functional across different runtimes and will continue as WASI evolves. However, the ongoing evolution of WASI, particularly the shift introduced with WASI 0.2+, has raised concerns about long-term stability. For many organizations, particularly those maintaining long-lived software, frequent runtime changes are not feasible.
+
+The meaningful, widespread use of WASI Preview 1 (P1) means that many runtimes and applications depend on its established behavior. Developers have invested significant time and effort into building applications, frameworks, and infrastructure around WASI P1, making it a critical part of the WebAssembly ecosystem. 
+
+There has been considerable effort expended to ensure that the WASI 0.1 tool chain remains useable while the WASI 0.2+ transition continues, an example of this is the [WASI Preview Transition](https://github.com/abrown/wasi-libc/blob/preview2-plan/docs/preview2-transition.md). There is as yet no way to run WASI 0.2+ binaries on a WASI 0.1 runtime, although there have been some investigations, like [this one from Marcin](https://github.com/loganek/wasi-snapshot-preview2-to-preview1-adapter), non have been successfully reached a conclusion and been used in production. Additionally, the long term continued investment in WASI 0.1 is not guaranteed. The organizations supporting WASI 0.1 are the same organizations driving WASI 0.2+, as WASI moves towards a 1.0 release these organizations will not have the need to continue to offer support for WASI 0.1.
```

Comment:
It might also be helpful to mention here the plan Luke laid out for `wasm32-wasip*-module`. Seems like that would be something realistic that p1-style runtimes could adopt for running newer binaries on older  runtimes?

Reply:
Or see what I suggested below...